presto-parser/src/test/java/com/facebook/presto/sql/TestExpressionFormatter.java

```diff
@@ -0,0 +1,247 @@
+import static org.testng.Assert.assertTrue;
+
+public class TestExpressionFormatter
+{
+    private static final SqlParser SQL_PARSER = new SqlParser();
+
+    @Test
+    public void testFormatStringLiteralPreservesNewlines()
+    {
+        String input = "line1\nline2\nline3";
+        String formatted = formatStringLiteral(input);
+
+        assertTrue(formatted.contains("\n"), "Newlines should be preserved in output");
+        assertEquals(formatted, "'line1\nline2\nline3'");
+    }
+
+    @Test
+    public void testFormatStringLiteralPreservesTabs()
+    {
+        String input = "col1\tcol2\tcol3";
+        String formatted = formatStringLiteral(input);
+
+        assertTrue(formatted.contains("\t"), "Tabs should be preserved in output");
+        assertEquals(formatted, "'col1\tcol2\tcol3'");
+    }
+
+    @Test
+    public void testFormatStringLiteralPreservesCarriageReturns()
+    {
+        String input = "line1\rline2";
+        String formatted = formatStringLiteral(input);
+
+        assertTrue(formatted.contains("\r"), "Carriage returns should be preserved in output");
+        assertEquals(formatted, "'line1\rline2'");
+    }
+
+    @Test
+    public void testFormatStringLiteralWithMixedWhitespace()
+    {
+        String input = "def foo():\n\treturn 'bar'\r\n";
+        String formatted = formatStringLiteral(input);
+
+        assertTrue(formatted.contains("\n"), "Newlines should be preserved");
+        assertTrue(formatted.contains("\t"), "Tabs should be preserved");
+        assertTrue(formatted.contains("\r"), "Carriage returns should be preserved");
+        assertEquals(formatted, "'def foo():\n\treturn ''bar''\r\n'");
+    }
+
+    @Test
+    public void testFormatStringLiteralWithPythonCode()
+    {
+        String pythonCode = "def process(x):\n" +
+                "\tif x > 0:\n" +
+                "\t\treturn x * 2\n" +
+                "\treturn 0";
+        String formatted = formatStringLiteral(pythonCode);
+
+        assertTrue(formatted.contains("\n"), "Newlines in Python code should be preserved");
+        assertTrue(formatted.contains("\t"), "Tabs in Python code should be preserved");
+        assertEquals(formatted, "'" + pythonCode + "'");
+    }
+
+    @Test
+    public void testFormatStringLiteralPreservesRegularStrings()
+    {
+        String input = "hello world";
+        String formatted = formatStringLiteral(input);
+        assertEquals(formatted, "'hello world'");
+    }
+
+    @Test
+    public void testFormatStringLiteralEscapesSingleQuotes()
+    {
+        String input = "it's a test";
+        String formatted = formatStringLiteral(input);
+        assertEquals(formatted, "'it''s a test'");
+    }
+
+    @Test
+    public void testFormatStringLiteralWithUnicodeCharacters()
+    {
+        String input = "hello \u0001 world";
+        String formatted = formatStringLiteral(input);
+
+        assertTrue(formatted.startsWith("U&'"), "Non-printable characters should use Unicode format");
```

Comment:
Um, I may understand wrong, but isn't your fix to solve this problem? Why still seeing U& in the formatted string?